omegaconf/base.py

```diff
@@ -265,6 +273,12 @@ def _key(self) -> Any:
     def _set_key(self, key: Any) -> None:
         self._metadata.key = key
 
+    def _is_flags_root(self) -> bool:
+        return self._metadata.flags_root
+
+    def _set_flags_root(self, flags_root: bool) -> None:
+        self._metadata.flags_root = flags_root
```

Comment:
I think this should also invalidate the flags cache if `flags_root` takes a new value

Reply:
yeah, good call.